ArcGISRuntimeSDKQt_CppSamples/Layers/ListKmlContents/ListKmlContents.cpp

```diff
@@ -210,6 +215,95 @@ void ListKmlContents::processSelectedNode(const QString& nodeName)
   }
 }
 
+void ListKmlContents::getViewpointFromKmlViewpoint(KmlNode* node)
+{
+  const KmlViewpoint kmlViewpoint = node->viewpoint();
+
+  if (!kmlViewpoint.isEmpty())
+  {
+    // altitude adjustment is needed for all but Absolute altitude mode
+    m_needsAltitudeFixed = (kmlViewpoint.altitudeMode() != KmlAltitudeMode::Absolute);
+
+    switch (kmlViewpoint.type())
+    {
+    case KmlViewpointType::LookAt:
+      m_viewpoint = Viewpoint(kmlViewpoint.location(), Camera(kmlViewpoint.location(), kmlViewpoint.range(),
+                                                              kmlViewpoint.heading(), kmlViewpoint.pitch(), kmlViewpoint.roll()));
+      return;
+    case KmlViewpointType::Camera:
+      m_viewpoint = Viewpoint(kmlViewpoint.location(), Camera(kmlViewpoint.location(),
+                                                              kmlViewpoint.heading(), kmlViewpoint.pitch(), kmlViewpoint.roll()));
+      return;
+    default:
+      qWarning("Unexpected KmlViewpointType");
+      return;
+    }
+  }
+
+  // if viewpoint was empty, then use node's extent
+  const Envelope nodeExtent = node->extent();
+  if (nodeExtent.isValid() && !nodeExtent.isEmpty())
+  {
+    // when no altitude is specified, assume elevation needs to be adjusted
+    m_needsAltitudeFixed = true;
+
+    if (nodeExtent.width() == 0 && nodeExtent.height() == 0)
+    {
+      // default values based on Google Earth
```

Comment:
not sure if we should say this?

Reply:
this comment was specified in the .NET implementation to indicate where the default values come from. Maybe I can change instead so it just says what the default values are? E.g., `\\ use default values: distance = 1000, pitch = 45` @lsmallwood

Reply:
sounds good to me